src/main/java/com/flightstats/hub/metrics/StatsDFilter.java

```diff
@@ -62,6 +65,21 @@ boolean isSecondaryReporting(String name) {
                 .isPresent();
     }
 
+    String extractName(String[] tags) {
+        try {
+            String namedMetric = Stream.of(tags)
+                    .filter((str -> StringUtils.isNotBlank(str) &&
+                            (str.contains("name") || str.contains("channel"))))
+                    .findAny()
+                    .orElse(null);
+            return namedMetric == null ?
```

Comment:
I think this might be more what you're looking for?
```
    String extractName(String[] tags) {
            return Stream.of(tags)
                    .filter(StringUtils::isNotBlank)
                    .filter(str -> str.contains("name") || str.contains("channel"))
                    .map(this::extractMetricName)
                    .findAny()
                    .orElse("");
    }

    private String extractMetricName(String fullMetricName) {
        try {
            return namedMetric.substring(namedMetric.indexOf(":") + 1);
        } catch (Exception ex) {
            return "";
        }
    }
```
That private method might need a better name, though.